File: JavaScript/climbStairs.js
/**
 * @param {number} n
 * @return {number}
 * time O(n) space O(1)
 */
var climbStairs = function(n) {

    if (n <= 2) {
        return n;
    }

    // // method 1 reccursion: may generate Time Limit Exceeded
    // else {
    //     return climbStairs(n - 1) + climbStairs(n - 2);
    // }

    // method 2 iteration
    var fib = 2,
        fibPre = 1,
        temp;
    for (i = 3; i <= n; i++) { // start from 2th fib, and ends at n, means nth fib number.
        temp = fib;
        fib = fib + fibPre;
        fibPre = temp;
    }
    return fib;

};

console.log(climbStairs(48));

File: Python/climbStairs.py
def climbStairsRecur(n):
        # write your code here
        if n == 1:
            return 1
        elif n == 2:
            return 2
        else:
            return climbStairsRecur(n-1)+climbStairsRecur(n-2)
            
            
def climbStairsIter(n):
        # write your code here
        if n <= 2:
            return n
        fib = 2
        fibPre = 1
        for i in range(2,n):
            temp = fib
            fib = fib + fibPre
            fibPre = temp
        return fib
        
print climbStairsIter(4)
print climbStairsRecur(4)